arch/x86_64/mcount-dynamic.c

```diff
@@ -369,8 +369,8 @@ static int patch_xray_func(struct mcount_dynamic_info *mdi, struct sym *sym)
 static void patch_code(struct mcount_dynamic_info *mdi,
 		       uintptr_t addr, uint32_t origin_code_size)
 {
-	unsigned char *origin_code_addr;
-	unsigned char call_insn[] = { 0xe8, 0x00, 0x00, 0x00, 0x00 };
+	char *origin_code_addr;
+	char call_insn[] = { 0xe8, 0x00, 0x00, 0x00, 0x00 };
```

Comment:
I don't know why clang declared arguments in `__builtin_clear_cache()` as `char *`.  To me, `char *` looks like a string.  So I use unsigned one when dealing with a binary data.

As the problem comes from the `origin_code_addr` what about making it `void *` instead and leave `call_insn` as is?

Reply:
I just changed it from `char*` to `void*`. Thanks.